### DISCUSSION TITLE
Anyone working on agentic framework or workflows using Spring AI?

### DISCUSSION THREAD

**Discussion by thesurlydev** [spring-projects/spring-ai#532] [General]
I'm already using Spring AI for some projects and curious if anyone has started looking at extending Spring AI in the context of an agentic framework or workflow. For example, frameworks such as Crew AI and Microsoft's Autogen use conversational agents with predefined tools to accomplish tasks as a group. 

Since Spring AI already has good abstractions, it seems like the next evolution is to extend it to compose groups of agents and tools.

Is anyone already doing this?

**Comment by rogerlzp** [comment]
I am doing that also.
agent, task, workflow, tools and other stuff maybe needed.
For workflow, there are existing softwares have the ability to build DAG, which may be helpful when building workflow.

**Comment by khaosans** [comment]
Just found spring ai, I've been in crewai and langchain so definitely looking for the workflow orchestration.  I'd also like to see those features here as well.

**Comment by coderphonui** [comment]
I am also working on a solution that using Spring AI for my own agent framework.

**Comment by escon1004** [comment]
I am also using Spring AI with Langgraph4J

**Comment by ivicac** [comment]
You can also check https://github.com/bytechefhq/bytechef
